Formula/mariadb@10.5.rb

```diff
@@ -36,7 +36,8 @@ class MariadbAT105 < Formula
   keg_only :versioned_formula
 
   # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-105/
-  deprecate! date: "2025-06-01", because: :unsupported
+  # and https://mariadb.org/about/#maintenance-policy
+  deprecate! date: "2025-06-24", because: :unsupported
```

Comment:
Use official EOL date from https://mariadb.org/about/#maintenance-policy
Release series | Stable (GA) Date | End-of-life
-- | -- | --
10.5 | 24 Jun 2020 | **24 Jun 2025**